### ISSUE TITLE
Can't install bun on Amazon Linux 2

### ISSUE THREAD

**Issue by timam** [oven-sh/bun#1043]
### Version

Not Available 

### Platform

Linux ip-*****.amzn2.x86_64 #1 SMP Tue Mar 29 23:15:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### What steps will reproduce the bug?

Install bun with install script on amazon linux 2. 
```
$ curl https://bun.sh/install | bash
```
Add following lines on .bashrc
```
export BUN_INSTALL="$HOME/.bun" 
export PATH="$BUN_INSTALL/bin:$PATH" 
```
Then try to execute bun command. 
```
$ bun
```

### How often does it reproduce? Is there a required condition?

100% of the time. 
OS: Amazon Linux 2

### What is the expected behavior?

Command should work, it should give following output
```
bun: a fast bundler, transpiler, JavaScript Runtime and package manager for web software.

  dev       ./a.ts ./b.jsx        Start a bun Dev Server
  bun       ./a.ts ./b.jsx        Bundle dependencies of input files into a .bun

  init                            Start an empty Bun project from a blank template
  create    next ./app            Create a new project from a template (bun c)
  run       test                  Run JavaScript with bun, a package.json script, or a bin
  install                         Install dependencies for a package.json (bun i)
  add       @compiled/react       Add a dependency to package.json (bun a)
  remove    browserify            Remove a dependency from package.json (bun rm)

  upgrade                         Get the latest version of bun
  completions                     Install shell completions for tab-completion
  discord                         Open bun's Discord server
  help                            Print this help menu

```

### What do you see instead?

```
bun: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by bun)
bun: /lib64/libc.so.6: version `GLIBC_2.27' not found (required by bun)
bun: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by bun)
bun: /lib64/libc.so.6: version `GLIBC_2.29' not found (required by bun)
```

### Additional information

After facing the issue I have installed Development Tools with, but no luck. my current glib version is 2.26
```
$ sudo yum groupinstall "Development Tools"
```

```
$ ldd --version
ldd (GNU libc) 2.26
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

**Comment by Jarred-Sumner** [comment]
This is caused by:
- glibc 2.27 requirement
- linux kernel >= 5.05 requirement

**Comment by rgoomar** [comment]
@timam which AMI are you using?

**Comment by timam** [comment]
ami-0f9e7e8f3cefd2cc0

**Comment by tobilg** [comment]
If I understand it correctly, the linux kernel >= 5.05 requirement is a showstopper when building a custom Lambda runtime for bun? Or would it potentially work when compiled on a Amazon Linux 2 with kernel version 4.14.x?

**Comment by Jarred-Sumner** [comment]
This is mostly fixed in Bun v0.3.0. Some functionality will not work until Amazon updates their Linux kernel from 4.x to 5.x or 6.x
